lib/entry/LiveEntry.js

```diff
@@ -212,8 +204,15 @@ LiveEntry.prototype.update = function(diagnosticsObj) {
     if (diagnosticsObj) {
         try {
             this.adapterDiagnosticsObj = diagnosticsObj;
-            saveFlavorDownloadersConfiguration.call(this);
-            this.diagnosticsAnalyzer.runFullDiagnosticsAnalysis(diagnosticsObj);
+            _.each(this.flavorsDownloaders, (f)=> {
+                f.setEncodedBitrate(this.adapterDiagnosticsObj.outputs[f.flavor]);
+            });
+            let entryInfo = {
+                'source' : this.adapterDiagnosticsObj,
+                'flavors' : JSON.stringify(this.flavorsDownloaders),
+                'playlistGenerator' : JSON.stringify(this.playlistGenerator)
+            };
+            this.diagnostics.handleDiagnosticsReport(entryInfo);
```

Comment:
note that here you are stringfing the flavors object and playlist generator.... this is not good, you should just call toJson, otherwsie you will work with strings and not object?